### PR TITLE
Fix: BlurNsfw not blurring embeds

### LINKED ISSUE #2837: [FEATURE REQUEST - BlurNSFW] - Blur Embedded Media
### Content

[The topic in the discord support](https://discord.com/channels/1015060230222131221/1026515880080842772/1280570489236095078)

Regarding the BlurNSFW, it would be ideal that it worked on linked images and embeds too.

At the moment it only blurs images(files) sent in the channel. If the image is embedded in a link or it's a pure image link from the web, the image is not blurred.

It would be interesting for the plugin to blur those embedded media too. Especially images.

Examples:
image
image

### Request Agreement

- [X] I have read the requirements for opening an issue above

### PR DESCRIPTION
**Pull request by rynmy**
Closes #2837 

demo:
image

### CODE REVIEW

File: src/plugins/blurNsfw/index.ts [hunk 1]
```diff
@@ -39,16 +39,16 @@ function setCss() {
...
-    authors: [Devs.Ven],
+    authors: [Devs.Ven, Devs.rya],
 
     patches: [
         {
             find: ".embedWrapper,embed",
             replacement: [{
-                match: /\.embedWrapper(?=.+?channel_id:(\i)\.id)/g,
-                replace: "$&+($1.nsfw?' vc-nsfw-img':'')"
+                match: /(?<=channel:\s*\i).*?\.container/g,
```
Comment (Masterjoona) [inline src/plugins/blurNsfw/index.ts:48]: You end up with a long match with this and you aren't capturing the channel variable

File: src/utils/constants.ts [hunk 2]
```diff
@@ -575,6 +575,10 @@ export const Devs = /* #__PURE__*/ Object.freeze({
         name: "RamziAH",
         id: 1279957227612147747n,
     },
+    rya: {
```
Comment (Masterjoona) [inline src/utils/constants.ts:578]: dont add yourself to devs for minor changes

Comment (ryanamay) [reply src/utils/constants.ts]: fixed in [93e1385](https://github.com/Vendicated/Vencord/pull/2862/commits/93e13858deb2110a7506b0539497d3f306a849ee)

**Comment by Nuckyz** [comment]
Isn't this just a fix of an already existing feature?

**Comment by ryanamay** [comment]
> Isn't this just a fix of an already existing feature?

yeah, though the issue is labeled as a feature request. which should i follow?

File: src/plugins/blurNsfw/index.ts [hunk 3]
```diff
@@ -45,10 +45,10 @@ export default definePlugin({
         {
             find: ".embedWrapper,embed",
             replacement: [{
-                match: /\.embedWrapper(?=.+?channel_id:(\i)\.id)/g,
-                replace: "$&+($1.nsfw?' vc-nsfw-img':'')"
+                match: /\.container(?=.+?a&&(this))/g,
```
[the remark this answers is not in the document]
Comment (ryanamay) [reply src/plugins/blurNsfw/index.ts]: attempted fix: [cd347c3](https://github.com/Vendicated/Vencord/pull/2862/commits/cd347c361d2c8f1342174976f3f20a20e3c25658) (removed `a` from regex match)

File: src/plugins/blurNsfw/index.ts [hunk 4]
```diff
@@ -45,10 +45,10 @@ export default definePlugin({
         {
             find: ".embedWrapper,embed",
             replacement: [{
-                match: /\.embedWrapper(?=.+?channel_id:(\i)\.id)/g,
-                replace: "$&+($1.nsfw?' vc-nsfw-img':'')"
+                match: /\.container/,
+                replace: "$&+(this.props.channel.nsfw? ' vc-nsfw-img': '')"
             }]
-        }
+        },
```
Comment (Nuckyz) [inline src/plugins/blurNsfw/index.ts:51]:
```suggestion
        }
```

**Review by Nuckyz — CHANGES_REQUESTED** [review]

**Comment by Nuckyz** [comment]
Thank you!

**Pull request merged**